File: app/src/main/java/com/mmjang/ankihelper/MyApplication.java
package com.mmjang.ankihelper;

import android.app.Activity;
import android.app.Application;
import android.content.Context;
import androidx.multidex.MultiDexApplication;
import com.jakewharton.threetenabp.AndroidThreeTen;
import com.mmjang.ankihelper.anki.AnkiDroidHelper;
//import com.tencent.bugly.crashreport.CrashReport;

import org.litepal.LitePalApplication;
import org.litepal.LitePal;
import okhttp3.OkHttpClient;
import androidx.work.WorkManager;
import org.litepal.LitePal; // Import for LitePal initialization

/**
 * Created by liao on 2017/4/27.
 */

public class MyApplication extends MultiDexApplication {
    private static Context context;
    private static Application application;
    private static AnkiDroidHelper mAnkiDroid;
    private static OkHttpClient okHttpClient;

    @Override
    public void onCreate() {
        super.onCreate();

        // Assign the application context here
        context = getApplicationContext();
        application = this;
        LitePal.initialize(this);
//        CrashReport.initCrashReport(getApplicationContext(), "398dc6145b", false);

    }

    // No changes needed here as you are returning 'context'
    public static Context getContext() {
        return context;
    }

    // No changes needed here as you are returning 'application'
    public static Application getApplication(){
        return application;
    }

    // Pass the Context as an argument
    public static AnkiDroidHelper getAnkiDroid(Context context) {
      if (mAnkiDroid == null) {
          mAnkiDroid = new AnkiDroidHelper(context);
      }
      return mAnkiDroid;
    }

    private static void getAnkiDroidPermission(Activity activity) {

    }

    public static OkHttpClient getOkHttpClient(){
        if(okHttpClient == null){
            okHttpClient = new OkHttpClient();
        }
        return okHttpClient;
    }
}

File: app/src/main/java/com/lmyby/ankiquicker/MyApplication.kt
package com.lmyby.ankiquicker

import android.app.Application
import android.content.Context
import androidx.multidex.MultiDexApplication
import androidx.work.WorkManager
import com.lmyby.ankiquicker.anki.AnkiDroidHelper
import com.lmyby.ankiquicker.data.history.HistoryUtil
import okhttp3.OkHttpClient

/**
 * Main Application class
 * Manages app-level singletons and initialization
 * Converted to Kotlin as part of Phase 13 final conversion
 */
class MyApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()

        // Assign the application context here
        context = applicationContext
        application = this

        // Initialize HistoryUtil for Room database in background to avoid blocking main thread
        Thread {
            HistoryUtil.initialize(applicationContext)
        }.start()

        // Language is now handled automatically by AndroidX AppCompatDelegate
        // No need for custom attachBaseContext or onConfigurationChanged

        // CrashReport.initCrashReport(getApplicationContext(), "398dc6145b", false)
    }

    companion object {
        private var context: Context? = null
        private var application: Application? = null
        private var mAnkiDroid: AnkiDroidHelper? = null
        private var okHttpClient: OkHttpClient? = null

        @JvmStatic
        fun getContext(): Context {
            return context ?: throw IllegalStateException("Application context is not initialized")
        }

        @JvmStatic
        fun getApplication(): Application {
            return application ?: throw IllegalStateException("Application is not initialized")
        }

        /**
         * Get AnkiDroidHelper instance (lazy initialized)
         * @param context Context to initialize helper with
         */
        @JvmStatic
        fun getAnkiDroid(context: Context): AnkiDroidHelper {
            if (mAnkiDroid == null) {
                mAnkiDroid = AnkiDroidHelper(context)
            }
            return mAnkiDroid!!
        }

        @JvmStatic
        fun getOkHttpClient(): OkHttpClient {
            if (okHttpClient == null) {
                okHttpClient = OkHttpClient()
            }
            return okHttpClient!!
        }
    }
}
